reproduce-figures/prepare-scpca-portal-data.R

```diff
@@ -0,0 +1,299 @@
+#
+#
+# ########################### Instructions ############################
+# 
+# Before running this script, you will need to take the following steps:
+#
+# 1. Ensure you have set up the `renv` environment in this project.
+#    For more information, refer to the top-level `README.md` file.
+# 2. Download all projects in the ScPCA Portal that are listed in the file `../sample-info/project-whitelist.txt`, with the following download options:
+#   - Ensure the selected modality is Single-cell
+#   - Ensure the selected Data Format is SingleCellExperiment (r)
+#   - Do _not_ click to merge all objects into the same sample
+#   - If the project contains multiplexed samples, you can exclude those samples (but it will not affect the code if they are included)
+# 3. In addition, you will need to download the merged version of project SCPCP000003, again in SingleCellExperiment format
+# 4. Without renaming downloaded files, place all downloaded project ZIP files a single directory for input to this script.
+#    Optionally, you can also store the merged version of SCPCP000003 in this directory, but it is not necessary.
+
+# Then, you can run this script as follows:
+# 
+#   Rscript prepare-scpca-portal-data.R \
+#     --portal_projects_dir <path to directory with all project zip files> \
+#     --merged_sce_path <path to merged SCE file>
+# 
+# By default, the output directories will be saved in the location in this repository where code expects them:
+# - `s3_files/` will be placed in the top level of the `scpca-paper-figures` repository
+# - `scpca-data/` will be placed inside the bulk analysis data directory, `scpca-paper-figures/pseudobulk-bulk-prediction/data/`
+# You can provide custom directories if you prefer; use the `--help` flag to see all script options.
+# Note that this script will not overwrite existing output directories without the `--overwrite` flag.
+
+
+
+renv::load()
+suppressPackageStartupMessages({
+  library(SingleCellExperiment)
+  library(optparse)
+})
+
+
+# Parse options --------
+option_list <- list(
+  make_option(
+    "--portal_projects_dir",
+    type = "character",
+    help = "Path to directory containing project ZIP files downloaded from the ScPCA Portal"
+  ),
+  make_option(
+    "--merged_sce_path",
+    type = "character",
+    help = "Path to the merged SCPCP000003 project ZIP file"
+  ),
+  make_option(
+    "--s3_files_dir",
+    type = "character",
+    default = here::here("s3_files"), 
+    help = "Output directory for `s3_files`. Default is in the top-level of the repository, which is where code expects it to be."
+  ),
+  make_option(
+    "--bulk_data_dir",
+    type = "character",
+    default = here::here("scpca-data"), #here::here("analysis", "pseudobulk-bulk-prediction", "data", "scpca-data"), 
+    help = "Output directory for bulk data. Default is `analysis/pseudobulk-bulk-prediction/data/scpca-data`, which is where code expects it to be."
```

Comment:
You have one default in `default` and mention a different one in `help`?

Reply:
this is some kind of copy/paste woopsie...